src/teacher.py

```diff
@@ -77,23 +77,30 @@ def _answer_Demonstration_query(
 
 
 @dataclass
-class TeacherInteractionMonitor(utils.Monitor):
-    """Wraps the interaction between agent and teacher to include generating
-    and answering queries."""
+class TeacherInteractionMonitor(utils.VideoMonitor):
```

Comment:
I'm not really a fan of this inheritance.. there's nothing inherent about a teacher that makes it have to be a VideoMonitor. This is one of those rare instances where multiple inheritance feels better. One idea would be:
```
class Monitor(abc.ABC):
  def observe(self, state: State, action: Optional[Action]) -> None:

class VideoMonitor(Monitor):
  def observe(self, state: State, action: Optional[Action]) -> None:
  def get_video(self) -> Video:

class TeacherInteractionMonitor(Monitor):
  (as before)

class TeacherInteractionMonitorWithVideo(TeacherInteractionMonitor, VideoMonitor):
  (an observe() method that calls both parents')
```
This might be messy though. Open to other thoughts as well.